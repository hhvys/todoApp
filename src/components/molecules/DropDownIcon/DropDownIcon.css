.div__dd__icon {
    background-color: inherit;
}

.div__dd__icon .icon-text {
    font-size: 10px;
    color: white;
}