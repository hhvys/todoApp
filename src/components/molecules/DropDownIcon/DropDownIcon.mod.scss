.icon {
  background-color: inherit;
  cursor: pointer;

}

.icon:hover {
  background-color: #769475;
}

.logo {
  fill: white;
  margin-top: 10px;
}

.iconText {
  font-size: 10px;
  color: white;
  font-weight: 500;
}
