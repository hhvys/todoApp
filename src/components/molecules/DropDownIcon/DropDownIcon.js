import React from 'react';
import './DropDownIcon.css';
import Symbol from "../../atoms/logos/Symbol";

class DropDownIcon extends React.Component {
	static defaultProps = {
		style: {
			height: 44,
			width: 52
		}
	};

	constructor(props) {
		super(props);
	}

	render() {
		const {
			iconSymbol,
			onClick,
			iconText,
			...props
		} = this.props;
		return (
			<div className={"div__dd__icon full-height d-flex align-items-center justify-content-center flex-column"}
					 onClick={onClick} {...props}>

				<div className={"icon-symbol full-height d-flex align-items-center justify-content-center"}>
					<Symbol symbolType={iconSymbol}
									style={{fill: 'white', marginTop: 10}}/>
				</div>

				<div className={"icon-text full-height d-flex align-items-center justify-content-center"}>
					{iconText}
				</div>

			</div>
		);
	}

}

export default DropDownIcon;