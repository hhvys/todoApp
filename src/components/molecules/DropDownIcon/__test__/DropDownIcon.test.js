import React from 'react';
import DropDownIcon from '../DropDownIcon';

describe('component', () => {
	describe('DropDownIcon', () => {
		it('should render', () => {
			expect(toJSON(shallow(<DropDownIcon/>))).toMatchSnapshot();
		});

		it('should pass iconSymbol', () => {
			const props = {
				iconSymbol: 'iconLogo'
			};
			expect(toJSON(shallow(<DropDownIcon {...props}/>))).toMatchSnapshot();
		});

		it('should pass iconText', () => {
			const props = {
				iconText: 'iconText'
			};
			expect(toJSON(shallow(<DropDownIcon {...props}/>))).toMatchSnapshot();
		});
	});
});
