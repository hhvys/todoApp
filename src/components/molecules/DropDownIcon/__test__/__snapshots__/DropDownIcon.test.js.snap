// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component DropDownIcon should pass iconSymbol 1`] = `
<div
  className="icon full-height d-flex align-items-center justify-content-center flex-column"
>
  <div
    className="full-height d-flex align-items-center justify-content-center"
  >
    <Icon
      className="logo"
      height={20}
      symbolType="iconLogo"
      width={20}
    />
  </div>
  <div
    className="iconText icon-text full-height d-flex align-items-center justify-content-center"
  />
</div>
`;

exports[`component DropDownIcon should pass iconText 1`] = `
<div
  className="icon full-height d-flex align-items-center justify-content-center flex-column"
>
  <div
    className="full-height d-flex align-items-center justify-content-center"
  >
    <Icon
      className="logo"
      height={20}
      width={20}
    />
  </div>
  <div
    className="iconText icon-text full-height d-flex align-items-center justify-content-center"
  >
    iconText
  </div>
</div>
`;

exports[`component DropDownIcon should render 1`] = `
<div
  className="icon full-height d-flex align-items-center justify-content-center flex-column"
>
  <div
    className="full-height d-flex align-items-center justify-content-center"
  >
    <Icon
      className="logo"
      height={20}
      width={20}
    />
  </div>
  <div
    className="iconText icon-text full-height d-flex align-items-center justify-content-center"
  />
</div>
`;
